plugins/optimization-detective/class-od-tag-visitor-context.php

```diff
@@ -48,6 +48,17 @@ final class OD_Tag_Visitor_Context {
 	 */
 	public $link_collection;
 
+	/**
+	 * Post ID for the od_url_metrics post which provided the URL Metrics in the collection.
+	 *
+	 * May be null if no post has been created yet.
+	 *
+	 * @since n.e.x.t
+	 * @var positive-int|null
+	 * @readonly
+	 */
+	public $url_metrics_post_id;
```

Comment:
I think we should omit the `post` from the name, as it can lead to confusion. Calling it `url_metrics_id` makes it clear it's the ID of the URL metrics, whereas if it contains "post", one could think this is referring to the post that lives at that URL.

It's okay to mention "post" when referring to it in combination with the `od_url_metrics` post type, because then it's obvious. But otherwise I'd refrain from it to avoid confusion.

```suggestion
	/**
	 * ID for the `od_url_metrics` post which provided the URL Metrics in the collection.
	 *
	 * May be null if no post has been created yet.
	 *
	 * @since n.e.x.t
	 * @var positive-int|null
	 * @readonly
	 */
	public $url_metrics_id;
```

Aside: Should it be `url_metrics_id` or `url_metric_id`?

Reply:
It should definitely not be `url_metric_id` since if you're talking about a single URL Metric then you would be using it's UUID. An `od_url_metrics` post always contains an array of URL Metrics, so this should be plural.

Reply:
> I think we should omit the `post` from the name, as it can lead to confusion. Calling it `url_metrics_id` makes it clear it's the ID of the URL metrics, whereas if it contains "post", one could think this is referring to the post that lives at that URL.

I'm not sure. The `od_url_metric_stored` action includes a context object that references `post_id`. In that case the `url_metrics_` prefix is not present, but that is more clear from context. For a tag visitor context, adding the prefix is needed because it wouldn't be clear from the tag visitor what the ID is referring to. But in both cases I thought it made sense to mention `post_id`. Probably the former should be updated to have the same as the latter for consistently. But back to the main point about whether to mention "post". I think it is helpful to mention 'post` so that the developer understands what kind of ID is being referred to and what you can do with it. Namely, you can use it to access the post meta. If it's just a bare `url_metrics_id` then the developer would need to look deeper to figure out what it is exactly.

Reply:
I think that it's a very common pattern in plugins to refer to this data as just "ID" (instead of "post ID") in combination with the plugin's specific purpose, so I'm not concerned that people wouldn't understand it's a post ID. Consider for example WooCommerce referring to a "product ID" - it's not saying "product post ID". It's so common that I think plugin developers would actually assume it's a post ID anyway. And by omitting the word "post" it's clear that it's the ID of the URL metrics themselves, not the post ID of the content in that URL.

Of course in theory it wouldn't hurt to include the word "post", but since URL metrics can also refer to a URL of a specific post, in this case that could be confusing. So IMO the downsides of using "URL metrics post ID" would be greater than the benefits.

Reply:
I've renamed `url_metrics_post_id` to `url_metrics_id` in deebfd7.

In 145f89e I've renamed `OD_URL_Metric_Store_Request_Context::$post_id` to `OD_URL_Metric_Store_Request_Context::$url_metrics_id` for consistency.